src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs

```diff
@@ -588,7 +588,8 @@ private static X509Certificate2 GenerateCertificate(X509KeyUsageFlags type, RSA
 
             X500DistinguishedName GetSubjectName()
             {
-                try { return new X500DistinguishedName("CN=" + (settings.RequestUrlHost ?? "localhost")); }
+                var host = settings.RequestUrlHosts.FirstOrDefault(host => host != "localhost");
+                try { return new X500DistinguishedName("CN=" + (host ?? "localhost")); }
```

Comment:
Can you format try .. cache block in your way

Reply:
Okay